device/simulation/tt_simulation_device.cpp

```diff
@@ -0,0 +1,261 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2023 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <vector>
+
+#include "common/logger.hpp"
+#include "common/assert.hpp"
+#include "device/driver_atomics.h"
+#include "device/tt_cluster_descriptor.h"
+
+#include "tt_simulation_device.h"
+#include "uv.h"
+
+tt_SimulationDevice::tt_SimulationDevice(const std::string &sdesc_path) : tt_device(sdesc_path){
+    // log_info(tt::LogEmulationDriver, "Instantiating simulation device");
+    soc_descriptor_per_chip.emplace(0, tt_SocDescriptor(sdesc_path));
+    std::set<chip_id_t> target_devices = {0};
+    
+    // Start VCS simulator in a separate process
+    TT_ASSERT(std::getenv("TT_METAL_VCS"), "TT_METAL_VCS not set, please provide path to the VCS binary");
+    uv_loop_t *loop = uv_default_loop();
+    uv_process_t child_p;
+    uv_process_options_t child_options = {0};
+
+    child_options.file = std::getenv("TT_METAL_VCS");
+    child_options.flags = UV_PROCESS_DETACHED;
+
+    int rv = uv_spawn(loop, &child_p, &child_options);
+    if (rv) {
+        log_info(tt::LogEmulationDriver, "Failed to spawn simulator process: {}", uv_strerror(rv));
```

Comment:
I think we have a log_fatal

Reply:
I just removed it and let the `TT_THROW` trigger instead if that works too